src/cgame/cg_animdelta.cpp

```diff
@@ -0,0 +1,115 @@
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Daemon; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+===========================================================================
+*/
+
+#include "cg_animdelta.h"
+
+namespace
+{
+
+int LoadDeltaAnimation( weapon_t weapon, const char* modelName, bool iqm )
+{
+	int handle = 0;
+	if ( iqm )
+	{
+		handle = trap_R_RegisterAnimation( va( "models/players/%s/%s.iqm:%s_delta", modelName, modelName, BG_Weapon( weapon )->name ) );
+	}
+	else
+	{
+		handle = trap_R_RegisterAnimation( va( "models/players/%s/%s_delta.md5anim", modelName, BG_Weapon( weapon )->name ) );
+	}
+	return handle;
+}
+
+}  // namespace
+
+bool AnimDelta::ParseConfiguration(clientInfo_t* ci, const char* token2, const char** data_p)
+{
+	if ( Q_stricmp( token2, "handBones" ) ) return false;
+	char* token = COM_Parse2( data_p );
+	if ( !token || token[0] != '{' )
+	{
+		Log::Notice( "^1ERROR^7: Expected '{' but found '%s' in %s's character.cfg", token, ci->modelName );
+		return false;
+	}
+	while ( 1 )
+	{
+		token = COM_Parse( data_p );
+		if ( !token || token[0] == '}' ) break;
+		boneIndicies_.push_back( trap_R_BoneIndex( ci->bodyModel, token ) );
+	}
+	return true;
+}
+
+bool AnimDelta::LoadData(clientInfo_t* ci)
+{
+	char newModelName[ MAX_QPATH ];
+	// special handling for human_(naked|light|medium)
+	if ( !Q_stricmp( ci->modelName, "human_naked"   ) ||
+		!Q_stricmp( ci->modelName, "human_light"   ) ||
+		!Q_stricmp( ci->modelName, "human_medium" ) )
+	{
+		Q_strncpyz( newModelName, "human_nobsuit_common", sizeof( newModelName ) );
+	}
+	else
+	{
+		Q_strncpyz( newModelName, ci->modelName, sizeof( newModelName ) );
+	}
+
+	refSkeleton_t base;
+	refSkeleton_t delta;
+	for ( int i = WP_NONE; i < WP_NUM_WEAPONS; ++i )
+	{
+		int handle = LoadDeltaAnimation( static_cast<weapon_t>( i ), newModelName, ci->iqm );
+		if ( !handle ) continue;
+		Log::Notice("Loaded delta for %s %s", newModelName, BG_Weapon( i )->humanName);
+		trap_R_BuildSkeleton( &delta, handle, 1, 1, 0, false );
+		// Derive the delta from the base stand animation.
+		trap_R_BuildSkeleton( &base, ci->animations[ TORSO_STAND ].handle, 1, 1, 0, false );
+		auto& weaponDeltas = deltas_[ i ];
+		weaponDeltas.resize( boneIndicies_.size() );
+		for ( size_t j = 0; j < boneIndicies_.size(); ++j )
+		{
+			VectorSubtract( delta.bones[ boneIndicies_[ j ] ].t.trans, base.bones[ boneIndicies_[ j ] ].t.trans, weaponDeltas[ j ].delta );
+			QuatInverse( base.bones[ boneIndicies_[ j ] ].t.rot );
+			QuatMultiply( base.bones[ boneIndicies_[ j ] ].t.rot, delta.bones[ boneIndicies_[ j ] ].t.rot, weaponDeltas[ j ].rot );
+		}
+	}
+	return true;
+}
+
+bool AnimDelta::Apply(const entityState_t* es, refSkeleton_t* skeleton)
```

Comment:
This returns "true" in all possible paths. This makes the return value essentially useless